### ISSUE TITLE
Vsock does not work on guest linux kernel version 6.1

### ISSUE THREAD

**Issue by kayoub-akamai** [firecracker-microvm/firecracker#4284]
# Describe the bug

When trying to use vsock on a guest with linux kernel version 6.1.63 there are infinite errors. It works with guest kernel version 5.10.202.

## To Reproduce
Used the vsock instructions at https://github.com/firecracker-microvm/firecracker/blob/main/docs/vsock.md#connecting-from-guest-to-host 
Config:
```
	"vsock": {
		"uds_path": "v.sock",
		"guest_cid": 3
	}
```
Command on host:
```
socat - UNIX-LISTEN:./v.sock_52
```
Command on vm:
```
socat - VSOCK-CONNECT:2:52
```
this results in a stream of errors from the vm that never ends:
```
2023-11-27T13:32:43.044305511 [anonymous-instance:main] vsock: RX queue error: BufDescMissing. Discarding the package.
2023-11-27T13:32:43.044308301 [anonymous-instance:main] vsock: RX queue error: BufDescMissing. Discarding the package.
2023-11-27T13:32:43.044310994 [anonymous-instance:main] vsock: RX queue error: BufDescMissing. Discarding the package.
2023-11-27T13:32:43.044313682 [anonymous-instance:main] vsock: RX queue error: BufDescMissing. Discarding the package.
2023-11-27T13:32:43.044316370 [anonymous-instance:main] vsock: RX queue error: BufDescMissing. Discarding the package.
2023-11-27T13:32:43.044319156 [anonymous-instance:main] vsock: RX queue error: BufDescMissing. Discarding the package.
and when I terminate the VM, the host socat is closed - which shows there was some interaction at least 
```
## Expected behaviour

It shouldn't have those errors.

## Environment

Firecracker versions tried 1.4 and 1.5
Host Kernel 5.15.0-88
Guest kernels that failed 6.4.7 and 6.1.63
Guest kernel that worked 5.10.202

## Additional context

Can't use vsock.

## Checks

Brought this up in Slack, was asked to make a ticket for it.

**Comment by roypat** [comment]
Hello @kayoub-akamai,

Thank you for your bug report. It seems that this is related to changes in the linux kernel's vsock driver originally merged for 6.2, but recently backported to 6.1 as well. We will look into fixing this issue as soon as possible.

**Comment by roypat** [comment]
Hi @kayoub-akamai, we have merged a fix for the vsock device when using recent Linux 6.1 builds as guests. Could you please check if you still encounter the issues described in this ticket using Firecracker compiled from `main`, and if yes, reopen this issue? Thank you!
